django/db/models/fields/related.py

```diff
@@ -1586,7 +1587,19 @@ class ForeignObject(RelatedField):
 
     def __init__(self, to, from_fields, to_fields, rel=None, related_name=None,
             related_query_name=None, limit_choices_to=None, parent_link=False,
-            on_delete=CASCADE, swappable=True, **kwargs):
+            on_delete=None, swappable=True, **kwargs):
+
+        if on_delete is None:
+            warnings.warn(
+                "on_delete will be a required kwarg for %s in Django 2.0. "
+                "See https://docs.djangoproject.com/en/%s/ref/models/fields/"
```

Comment:
Do you think it would be worth adding "Set it to models.CASCADE if you want to maintain the current default behavior."?